### PR TITLE
Plugin peerDependencies

### LINKED ISSUE #3575: Add peerDependencies to every plugin
Going forward, we'll be doing every so often major releases for Gatsby but individual plugins might not need upgraded as the APIs they depend on won't have changed. We should prepare for that now by adding a `peerDependencies` for Gatsby to every plugin. Then when there is a breaking change that affects the plugin or there's a new major release of Gatsby, we update the peerDependencies for the plugin.

### PR DESCRIPTION
**Pull request by tsriram**
Closes #3575

### CODE REVIEW

File: packages/gatsby-plugin-catch-links/package.json [hunk 1]
```diff
@@ -1,17 +1,15 @@
 {
   "name": "gatsby-plugin-catch-links",
   "version": "1.0.14",
-  "description": "Intercepts local links from markdown and other non-react pages and does a client-side pushState to avoid the browser having to refresh the page.",
+  "description":
```
[the remark this answers is not in the document]
Comment (tsriram) [reply packages/gatsby-plugin-catch-links/package.json]: Guess the prettier extension in my editor did this. Running `yarn format` doesn't change this back to single line either. Should I manually move this to single line?

[the remark this answers is not in the document]
Comment (KyleAMathews) [reply packages/gatsby-plugin-catch-links/package.json]: Eh, it's not a big deal. We're not running prettier on JSON files yet which is why this happened 👍

[the remark this answers is not in the document]
Comment (KyleAMathews) [reply packages/gatsby-plugin-catch-links/package.json]: And if Prettier wants it, I want it :-)

**Comment by KyleAMathews** [comment]
Thanks!

**Comment by jlengstorf** [comment]
Hiya @tsriram! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**